app/Http/Book/BookRepository.php

```diff
@@ -68,28 +68,31 @@ public function uploadBookCover($file, Book $book)
     {
 
         // if book create
-        if ($book && $file) {
-
-            // get the folder for storage
-            $destination = '/books/'.$book->id.'/';
-
-            // get a clean filename
-            $filename = str_random(16).'.'.$file->getClientOriginalExtension();
-
-            // absolute path
-            $absolute_path = $destination.$filename;
-
-            // move the file to the destination  file
-            $storage = \Storage::put($absolute_path, file_get_contents($file->getRealPath()));
-
-            // return filename if file is moved
-            if ($storage) {
-
-                // store the filename
-                $book->cover_url = $filename;
-                $book->save();
-
-                return $filename;
+        if ($book) {
+
+            if($file) {
+
+                // get the folder for storage
+                $destination = '/books/'.$book->id.'/';
+    
+                // get a clean filename
+                $filename = str_random(16).'.'.$file->getClientOriginalExtension();
+    
+                // absolute path
+                $absolutePath = $destination.$filename;
+    
+                // move the file to the destination  file
+                $storage = \Storage::put($absolute_path, file_get_content($file->getRealPath()));
```

Comment:
You have a typo in your method call. It should be `file_get_contents` instead of `file_get_content`.